lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1295,8 +1298,9 @@ class MuxPad : public mlir::RewritePattern {
     if (width < 0)
       return failure();
 
-    auto pad = [&](FIRRTLBaseValue input) -> Value {
-      auto inputWidth = input.getType().getBitWidthOrSentinel();
+    auto pad = [&](Value input) -> Value {
+      auto inputWidth = firrtl::type_cast<FIRRTLBaseType>(input.getType())
```

Comment:
This could be `TypedValue<BaseTypeAliasor<FIRRTLBaseType>>` (perhaps a new "using" since AFAICT we basically never have a `TypedValue<FIRRTLBaseType>` anymore?) to keep the requirement in the interface instead of asserting the value is "of the right type" but I never got to rooting out that pattern in the code base so maybe this is fine.